components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +78,19 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public String resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        if (resetRoute != null) {
+            resetRoute.reset(true);
+        } else {
+            String description = "The provided route id does not exists";
+            ResponseEntity<?> objectResponseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT).body(description);
+            ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Not a problem, but as mentioned previously, we should use HTTP 404 to indicate a client error.